### PR TITLE
[ISSUE #367] Optimize plugin load

### PR DESCRIPTION
**Pull request by ruanwenjun**
1. Optimize plugin load, support load plugin by plugin instance name, these means we needn't load all plugin
2. Optimize install plugin, suppory install plugin by ./gradlew jar dist
3. Make UrlClassLoader siglenton

### CODE REVIEW

File: eventmesh-connector-plugin/build.gradle [hunk 1]
```diff
@@ -13,25 +13,4 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
- */
-
-task copyConnectorPlugin(dependsOn: ['jar']) {
-    doFirst {
-        new File(projectDir, '../eventmesh-connector-plugin/dist/apps').mkdir()
```
Comment (qqeasonchen) [inline eventmesh-connector-plugin/build.gradle:20]: may not need dist directory

File: eventmesh-connector-plugin/build.gradle [hunk 2]
```diff
@@ -13,25 +13,4 @@
...
-        new File(projectDir, '../dist/plugin/connector').mkdirs()
```
Comment (qqeasonchen) [inline eventmesh-connector-plugin/build.gradle:21]: may not need dist directory, just $HOME/plugin/connector

File: eventmesh-runtime/build.gradle [hunk 3]
```diff
@@ -24,14 +24,14 @@ dependencies {
     api 'io.prometheus:simpleclient_httpserver'
 
     implementation project(":eventmesh-connector-plugin:eventmesh-connector-api")
-    implementation project(":eventmesh-connector-plugin:eventmesh-connector-standalone")
+    implementation project(":eventmesh-connector-plugin:eventmesh-connector-rocketmq")
```
Comment (qqeasonchen) [inline eventmesh-runtime/build.gradle:27]: Set default connector as standalone is ok in my opinion.

File: eventmesh-security-plugin/build.gradle [hunk 4]
```diff
@@ -13,25 +13,4 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
- */
-
-task copyAclPlugin(dependsOn: ['jar']) {
-    doFirst {
-        new File(projectDir, '../eventmesh-security-plugin/dist/apps').mkdir()
-        new File(projectDir, '../dist/plugin/security').mkdirs()
```
Comment (qqeasonchen) [inline eventmesh-security-plugin/build.gradle:21]: also, do not need dist directory.

File: eventmesh-spi/src/main/java/org/apache/eventmesh/spi/loader/JarExtensionClassLoader.java [hunk 5]
```diff
@@ -46,26 +47,31 @@
     private static final ConcurrentHashMap<Class<?>, Map<String, Class<?>>> EXTENSION_CLASS_CACHE =
             new ConcurrentHashMap<>(16);
 
+    // ./dist/plugin
     private static final String EVENTMESH_EXTENSION_PLUGIN_DIR = System.getProperty("eventMeshPluginDir",
-            "./dist/plugin");
+            Paths.get(".", "dist", "plugin").toString());
```
Comment (qqeasonchen) [inline eventmesh-spi/src/main/java/org/apache/eventmesh/spi/loader/JarExtensionClassLoader.java:52]: also, do not need dist directory.

File: eventmesh-connector-plugin/build.gradle (hunk 2 above)
Comment (ruanwenjun) [reply eventmesh-connector-plugin/build.gradle:21]: Now this task has been removed, we can use the `dist` task to insatll all plugin, the plugin directory will look like
image

File: eventmesh-runtime/build.gradle (hunk 3 above)
Comment (ruanwenjun) [reply eventmesh-runtime/build.gradle:27]: Agree with you, done.

File: eventmesh-spi/src/main/java/org/apache/eventmesh/spi/loader/JarExtensionClassLoader.java (hunk 5 above)
Comment (ruanwenjun) [reply eventmesh-spi/src/main/java/org/apache/eventmesh/spi/loader/JarExtensionClassLoader.java:52]: Yes, has removed the `dist`.

**Review by qqeasonchen — APPROVED** [review]

**Comment by qqeasonchen** [comment]
@ruanwenjun please help to fix it, when execute build.

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/easonchen/git/incubator-eventmesh/build.gradle' line: 215

* What went wrong:
Could not compile build file '/Users/easonchen/git/incubator-eventmesh/build.gradle'.
> startup failed:
  build file '/Users/easonchen/git/incubator-eventmesh/build.gradle': 215: expecting ')', found ',' @ line 215, column 46.
     ginTypeMap.forEach((pluginType, pluginIn
                                   ^
  
  1 error

**Pull request merged**